testsuite/tests/singlecluster/gateway/reconciliation/test_update_ratelimitpolicy_target_ref.py

```diff
@@ -0,0 +1,110 @@
+
+pytestmark = [pytest.mark.kuadrant_only]
+
+
+@pytest.fixture(scope="module")
+def wildcard_domain2(base_domain):
+    """Wildcard domain for Gateway B"""
+    return f"*.{base_domain}"
+
+
+@pytest.fixture(scope="module")
+def gateway(request, kuadrant, cluster, blame, wildcard_domain, module_label):  # pylint: disable=unused-argument
+    """Create and configure Gateway A"""
+    gw = KuadrantGateway.create_instance(cluster, blame("gw"), {"app": module_label})
+    gw.add_listener(GatewayListener(hostname=wildcard_domain))
+    request.addfinalizer(gw.delete)
+    gw.commit()
+    gw.wait_for_ready()
+    return gw
+
+
+@pytest.fixture(scope="module")
+def gateway_b(request, kuadrant, cluster, blame, wildcard_domain2, module_label):  # pylint: disable=unused-argument
+    """Create and configure Gateway B"""
+    gw = KuadrantGateway.create_instance(cluster, blame("gw-b"), {"app": module_label})
+    gw.add_listener(GatewayListener(hostname=wildcard_domain2))
+    request.addfinalizer(gw.delete)
+    gw.commit()
+    gw.wait_for_ready()
+    return gw
+
+
+@pytest.fixture(scope="module")
+def hostname_b(gateway_b, exposer, blame) -> Hostname:
+    """Expose Hostname for Gateway B"""
+    hostname = exposer.expose_hostname(blame("hostname-b"), gateway_b)
+    return hostname
+
+
+@pytest.fixture(scope="module")
+def route_b(
+    request, gateway_b, blame, hostname_b, module_label, backend
+) -> GatewayRoute:  # pylint: disable=unused-argument
+    """Create and configure Route B"""
+    route = HTTPRoute.create_instance(gateway_b.cluster, blame("route-b"), gateway_b, {"app": module_label})
+    route.add_hostname(hostname_b.hostname)
+    route.add_backend(backend)
+    request.addfinalizer(route.delete)
+    route.commit()
+    route.wait_for_ready()
+    return route
+
+
+@pytest.fixture(scope="module")
+def client_b(route_b, hostname_b):  # pylint: disable=unused-argument
+    """Returns httpx client for Gateway B"""
+    client = hostname_b.client()
+    yield client
+    client.close()
+
+
+@pytest.fixture(scope="module")
+def rate_limit_policy(request, cluster, blame, module_label, gateway, route_b):  # pylint: disable=unused-argument
+    """RateLimitPolicy for testing"""
+    policy = RateLimitPolicy.create_instance(cluster, blame("limit"), gateway, labels={"testRun": module_label})
+    policy.add_limit("basic", [Limit(5, "10s")])
+    request.addfinalizer(policy.delete)
+    policy.commit()
+    return policy
+
+
+@pytest.fixture(scope="module")
+def dns_policy_b(blame, gateway_b, module_label, dns_provider_secret, request):
+    """DNSPolicy fixture for Gateway B"""
+    policy = DNSPolicy.create_instance(
+        gateway_b.cluster, blame("dns-b"), gateway_b, dns_provider_secret, labels={"app": module_label}
+    )
+    request.addfinalizer(policy.delete)
+    policy.commit()
+    policy.wait_for_ready()
+    return policy
+
+
+def test_update_ratelimit_policy_target_ref(
```

Comment:
Now that you have divided these tests, authpolicy doesn't need to be used here anymore. See if you can remove it from this test and save some time during the setup